backend/routes/api/spots.js

```diff
@@ -0,0 +1,160 @@
+const express = require('express');
+const { Op } = require('sequelize');
+const bcrypt = require('bcryptjs');
+const { check } = require('express-validator');
+const { handleValidationErrors } = require('../../utils/validation');
+const { Sequelize } = require('sequelize')
+
+const { setTokenCookie, requireAuth } = require('../../utils/auth');
+const { Spot } = require('../../db/models');
+const { SpotImage } = require('../../db/models');
+const { User } = require('../../db/models');
+const { Review } = require('../../db/models');
+
+
+const router = express.Router();
+
+const avgStarRating = (reviews) => {
+
+}
+
+const validateSpot = [
+    check('address')
+        .exists({ checkFalsy: true})
+        .withMessage('Street address is required'),
+    check('city')
+        .exists({ checkFalsy: true})
+        .withMessage('City is required'),
+    check('state')
+        .exists({ checkFalsy: true})
+        .withMessage('State is required'),
+    check('country')
+        .exists({ checkFalsy: true})
+        .withMessage('Country is required'),
+    check('lat')
+        .exists({ checkFalsy: true})
+        .withMessage('Latitude must be within -90 and 90'),
+    check('lng')
+        .exists({ checkFalsy: true})
+        .withMessage('Longitude must be within -180 and 180'),
+    check('name')
+        .exists({ checkFalsy: true})
+        .withMessage('Name must be less than 50 characters'),
+    check('description')
+        .exists({ checkFalsy: true})
+        .withMessage('Description is required'),
+    check('price')
+        .exists({ checkFalsy: true})
+        .withMessage('Price per day must be a positive number'),
+    handleValidationErrors
+];
+
+
+router.get('/:spotId', async (req, res) => {
+    const spotid = parseInt(req.params.spotId);
+    
+    
+   const spot = await Spot.findOne({
+        where: {id: spotid},
+        attributes: {
+            include: [
+              [
+                Sequelize.fn('COUNT', Sequelize.col('Reviews.id')),
+                'numReviews',
+              ],
+              [
+                Sequelize.fn('AVG', Sequelize.col('Reviews.stars')),
+                'avgStarRating',
+              ],
+            ],
+          },
```

Comment:
I think this is unnecessary because you already set the Spots model to load `numReviews` and `avgStarRating` in `/models/spot.js`